### PR TITLE
Update framework docs / phpdoc for versioning changes

### PR DESCRIPTION
**Pull request by tractorcow**
Parent story: https://github.com/silverstripe/silverstripe-framework/issues/7359

Also adds a necessary extension point to GridFieldDetailForm_ItemRequest

### CODE REVIEW

File: docs/en/02_Developer_Guides/00_Model/10_Versioning.md [hunk 1]
```diff
@@ -3,13 +3,13 @@ summary: Add versioning to your database content through the Versioned extension
 
 # Versioning
 
-Database content in SilverStripe can be "staged" before its publication, as well as track all changes through the 
+Database content in SilverStripe can be "staged" before its publication, as well as track all changes through the
```
Comment (flamerohr) [inline docs/en/02_Developer_Guides/00_Model/10_Versioning.md]: I'd be careful of trimming trailing whitespaces in markdown - as two spaces translate to a `<br>` in html

File: docs/en/02_Developer_Guides/00_Model/10_Versioning.md [hunk 2]
```diff
@@ -308,7 +309,19 @@ class Banner extends Page
...
+
+Ownership can be used with non-versioned dataobjects, as the necessary functionality is included by default
+by the versioned object through the `[api:SilverStripe\Versioned\RecursivePublishable]` extension which is
+applied to all objects.
+
+However, it is important to note that even when saving un-versioned objects, it is necessary to use
+`->publishRecursive()` to trigger a recursive publish.
+
+As long as the `owns` config is specified on all elements in the tree, you can use any combination of
+versioned or unversioned dataobjects.
```
Comment (flamerohr) [inline docs/en/02_Developer_Guides/00_Model/10_Versioning.md]: Does this mean that we don't need to worry about `->publishRecursive()` if we have an `owns` defined?

Maybe elaborate that the `owns` config behaves in similar ways to the `owns` config in a versioned object?

File: docs/en/02_Developer_Guides/00_Model/10_Versioning.md (hunk 1 above)
Comment (tractorcow) [reply docs/en/02_Developer_Guides/00_Model/10_Versioning.md]: Well we have no need for a `<br>` here so it should be ok.

Comment (flamerohr) [reply docs/en/02_Developer_Guides/00_Model/10_Versioning.md]: yeah, I don't want to spend time going through all the whitespace changes to see if you did end up trimming a double space at the end :P

it's pretty much "please don't trim for markdown"

**Comment by flamerohr** [comment]
Thanks for reverting the trimming in markdown

Can merge on green.

File: docs/en/02_Developer_Guides/00_Model/10_Versioning.md (hunk 1 above)
Comment (dhensby) [reply docs/en/02_Developer_Guides/00_Model/10_Versioning.md]: > I'd be careful of trimming trailing whitespaces in markdown - as two spaces translate to a <br> in html

TBH, if people want a break, they should be using a double new line not two spaces - MD should be readable in both it's raw form and it's rendered form.

Comment (flamerohr) [reply docs/en/02_Developer_Guides/00_Model/10_Versioning.md]: that's true, but two new lines means a new paragraph block right?

**Pull request merged**